__tests__/table_moveColumn.js

```diff
@@ -0,0 +1,476 @@
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(
+              td({ rowspan: 2 }, p('M1')),
+              td(p('0')),
+              td(p('1')),
+              td(p('2')),
+              td(p('3'))
+            ),
+            row(
+              td({ colspan: 2 }, p('4')),
+              td({ rowspan: 2, colspan: 2 }, p('M2'))
+            ),
+            row(td(p('5')), td(p('6')), td(p('7')))
+          )
+        )
+      );
+
+      const expectedResult = doc(
+        table(
+          row(
+            td({ rowspan: 2 }, p('M1')),
+            td(p('2')),
+            td(p('3')),
+            td(p('0')),
+            td(p('1'))
+          ),
+          row(
+            td({ rowspan: 2, colspan: 2 }, p('M2')),
+            td({ colspan: 2 }, p('4'))
+          ),
+          row(td(p('5')), td(p('6')), td(p('7')))
+        )
+      );
+
+      it('should move column 1 to 3', () => {
+        let newTr = moveColumn(1, 3)(tr);
+        expect(newTr.doc).toEqualDocument(expectedResult);
+      });
+
+      it('should move column 2 to 3', () => {
+        let newTr = moveColumn(2, 3)(tr);
+        expect(newTr.doc).toEqualDocument(expectedResult);
+      });
+
+      it('should move column 2 to 4', () => {
+        let newTr = moveColumn(2, 4)(tr);
+        expect(newTr.doc).toEqualDocument(expectedResult);
+      });
+
+      it('should move column 3 to 1', () => {
+        let newTr = moveColumn(3, 1)(tr);
+        expect(newTr.doc).toEqualDocument(expectedResult);
+      });
+
+      it('should move column 3 to 2', () => {
+        let newTr = moveColumn(3, 2)(tr);
+        expect(newTr.doc).toEqualDocument(expectedResult);
+      });
+
+      it('should move column 4 to 1', () => {
+        let newTr = moveColumn(4, 1)(tr);
+        expect(newTr.doc).toEqualDocument(expectedResult);
+      });
+
+      it('should move column 4 to 2', () => {
+        let newTr = moveColumn(4, 2)(tr);
+        expect(newTr.doc).toEqualDocument(expectedResult);
+      });
+    });
+
+    describe('table 3x4', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(
+              td({ rowspan: 2 }, p('M1')),
+              td(p('0')),
+              td(p('1')),
+              td(p('2'))
+            ),
+            row(td({ colspan: 2 }, p('3')), td({ rowspan: 2 }, p('m2'))),
+            row(td(p('4')), td(p('5')), td(p('6')))
+          )
+        )
+      );
+
+      const expectedResult = doc(
+        table(
+          row(td(p('0')), td(p('1')), td({ rowspan: 2 }, p('M1')), td(p('2'))),
+          row(td({ colspan: 2 }, p('3')), td({ rowspan: 2 }, p('m2'))),
+          row(td(p('5')), td(p('6')), td(p('4')))
+        )
+      );
+
+      it('should move column 0 to 2', () => {
+        let newTr = moveColumn(0, 2)(tr);
+        expect(newTr.doc).toEqualDocument(expectedResult);
+      });
+
+      it('should move column 0 to 1', () => {
```

Comment:
should throw because of the col-spanning cell `td({ colspan: 2 }, p('3'))`

Reply:
same as above